corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -22,3 +28,27 @@ def force_update_cleanliness_flags():
     # that there are no bugs in the weekly task.
     # If we haven't seen any issues by the end of 2015 (so 6 runs) we should remove this.
     set_cleanliness_flags_for_all_domains(force_full=True)
+
+
+@task(queue=ASYNC_RESTORE_QUEUE)
+def get_async_restore_payload(restore_config):
+    """Process an async restore
+    """
+    response = restore_config.generate_payload(async_task=current_task)
+
+    # delete the task id from the task, since the payload can now be fetched from the cache
+    restore_config.cache.delete(restore_config.async_cache_key)
+
+    return response
+
+
+@after_task_publish.connect
+def update_celery_state(sender=None, body=None, **kwargs):
+    """Used to update celery state so we know if an async restore made it to celery
```

Comment:
thanks for the explanation, that makes sense. i'm not sure that that bug would happen, i.e. i'd hope celery would make the task as failed if someone hard restarted celery during processing of a task. since this function is very important, i'm in favor of keeping it just for robustness and unambiguity. (agree maybe a bit more in the docstring would be helpful)